### DISCUSSION TITLE
Dot Notation

### DISCUSSION THREAD

**Discussion by jedi1992** [EvgSkv/logica#107] [Q&A] [answered]
Hi,
In this example https://twitter.com/EvgSkvDev/status/1387171962963652608

a) what are the main predicates ?
b) Are the  tickets itself a list in the Purchases ? Also it says ticket[dot]tier  . I guess this is the first time dotted notation has come in.  Or is it some kind of a list of structures.
c) Also if a column is a list will it make sense to capture it in the column name with an indication like tickets[]

Thanks.

**Comment by EvgSkv** [comment, marked as answer]
Hello!
a) I am not sure I understand what do you mean by "main predicates". It is a definition of PremiumPurchases predicate that assumes that predicate Purchases is already defined.
b) Yes, `tickets` is a list of structures. Dot notation is used to address fields of structures.
c) I am not aware of a language that would use `[]` suffix in the name of a list variable. Indeed C/C++ use `[]` suffix for denoting that variable is a type. But it's part of type definition syntax and at this moment Logica doesn't have explicit typing.

I've submitted a running [CoLab](https://github.com/EvgSkv/logica/blob/main/examples/more/List_Comprehension.ipynb) example.

Let me know if you have further questions.

**Reply by jedi1992** [reply]
Perfect. Thanks.
